test/e2e/apps/replica_set.go

```diff
@@ -524,3 +534,138 @@ func testRSLifeCycle(f *framework.Framework) {
 
 	framework.ExpectNoError(err, "failed to see replicas of %v in namespace %v scale to requested amount of %v", rs.Name, ns, rsPatchReplicas)
 }
+
+func testRSStatus(f *framework.Framework) {
+	ns := f.Namespace.Name
+	c := f.ClientSet
+	rsClient := c.AppsV1().ReplicaSets(ns)
+
+	// Define ReplicaSet Labels
+	rsPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  WebserverImageName,
+	}
+	labelSelector := labels.SelectorFromSet(rsPodLabels).String()
+	framework.Logf("labels: %v", labelSelector)
+
+	rsName := "test-rs"
+	replicas := int32(1)
+
+	w := &cache.ListWatch{
+		WatchFunc: func(options metav1.ListOptions) (watch.Interface, error) {
+			options.LabelSelector = labelSelector
+			return rsClient.Watch(context.TODO(), options)
+		},
+	}
+	rsList, err := c.AppsV1().ReplicaSets("").List(context.TODO(), metav1.ListOptions{LabelSelector: labelSelector})
+	framework.ExpectNoError(err, "failed to list Replicasets")
+
+	ginkgo.By("Create a Replicaset")
+	rs := newRS(rsName, replicas, rsPodLabels, WebserverImageName, WebserverImage, nil)
+	testReplicaSet, err := c.AppsV1().ReplicaSets(ns).Create(context.TODO(), rs, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+
+	ginkgo.By("Verify that the required pods have come up.")
+	err = e2epod.VerifyPodsRunning(c, ns, "sample-pod", false, replicas)
+	framework.ExpectNoError(err, "Failed to create pods: %s", err)
+
+	ginkgo.By("Getting /status")
+	rsResource := schema.GroupVersionResource{Group: "apps", Version: "v1", Resource: "replicasets"}
+	rsStatusUnstructured, err := f.DynamicClient.Resource(rsResource).Namespace(ns).Get(context.TODO(), rsName, metav1.GetOptions{}, "status")
+	framework.ExpectNoError(err, "Failed to fetch the status of replicaset %s in namespace %s", rsName, ns)
+	rsStatusBytes, err := json.Marshal(rsStatusUnstructured)
+	framework.ExpectNoError(err, "Failed to marshal unstructured response. %v", err)
+
+	var rsStatus appsv1.ReplicaSet
+	err = json.Unmarshal(rsStatusBytes, &rsStatus)
+	framework.ExpectNoError(err, "Failed to unmarshal JSON bytes to a replicaset object type")
+	framework.Logf("Replicaset %s has Conditions: %v", rsName, rsStatus.Status.Conditions)
+
+	ginkgo.By("updating the Replicaset Status")
```

Comment:
We usually expect the controller to update the status not by other actor. So, I am wondering if we are just hitting this to increase the coverage where as we in reality this cannot and shouldn't happen

Reply:
I am wondering if you can mimic the patch by modifying the spec and watch if the status subresource has been patched instead of manually updating the status

Reply:
The process of checking `/status` endpoints has gone through a number of discussions in conformance meetings for [Service Status endpoints](https://github.com/kubernetes/kubernetes/issues/94867) and [API Service Status](https://github.com/ii/kubernetes/blob/apiservice-status-endpoints/test/e2e/apimachinery/aggregator.go#L546-L670). 

I'm trying to have a clean common pattern for testing any `/status` endpoint no matter the resource.